bootstrap/kubeadm/controllers/kubeadmconfig_controller_test.go

```diff
@@ -193,6 +194,50 @@ func TestKubeadmConfigReconciler_Reconcile_ReturnEarlyIfMachineHasBootstrapData(
 	}
 }
 
+func TestKubeadmConfigReconciler_ReturnEarlyIfClusterInfraNotReady(t *testing.T) {
+	cluster := newCluster("cluster")
+	machine := newMachine(cluster, "machine")
+	config := newKubeadmConfig(machine, "cfg")
+
+	//cluster infra not ready
+	cluster.Status = clusterv1.ClusterStatus{
+		InfrastructureReady: false,
+	}
+
+	objects := []runtime.Object{
+		cluster,
+		machine,
+		config,
+	}
+	myclient := fake.NewFakeClientWithScheme(setupScheme(), objects...)
+
+	k := &KubeadmConfigReconciler{
+		Log:    log.Log,
+		Client: myclient,
+	}
+
+	request := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: "default",
+			Name:      "cfg",
+		},
+	}
+
+	expectedResult := reconcile.Result{}
+	actualResult, actualError := k.Reconcile(request)
+
+	if actualResult.Requeue != expectedResult.Requeue {
+		t.Fatalf("TestKubeadmConfigReconciler_ReturnEarlyIfClusterInfraNotReady failed, reconcile result doesn't match, Want %v, Got %v", expectedResult, actualResult)
```

Comment:
You don't need to include the name of the test function in the error message. Also you can use Errorf instead of Fatalf